prow/cmd/deck/main.go

```diff
@@ -1663,3 +1670,40 @@ func defaultLensRemoteConfig(lfc *config.LensFileConfig) error {
 
 	return nil
 }
+
+// ValidatePath validates various parts (e.g. bucket, folder, etc.) of a storage path.
+func ValidatePath(cfg config.Getter, path string) error {
+	if !cfg().Deck.EnableWhitelist {
+		return nil
+	}
+
+	parts := strings.SplitN(path, "/", 4)
+	if len(parts) < 4 {
+		return fmt.Errorf("invalid path: %s expected <key-type>/<bucket>/<folder>/<sub-folder...>", path)
+	}
+
+	// Check bucket
+	bucket := parts[1]
+	bwl := sets.String{}
+	for _, dc := range cfg().Plank.DefaultDecorationConfigs {
+		bwl.Insert(dc.GCSConfiguration.Bucket)
```

Comment:
I think it would be better if we also included buckets specified in job specific decoration config so they don't have to be included in `AdditionalBuckets`. I'm not sure if that is worthwhile or quick enough to check though.

Reply:
The major downside is that on _every_ validated request each job will need to be traversed. Also, it cannot really be cached since any precomputed values will not include subsequent config updates. Let me investigate the feasibility.